grails-app/controllers/org/pih/warehouse/api/PutawayApiController.groovy

```diff
@@ -40,11 +42,15 @@ class PutawayApiController {
         if (!order) {
             throw new IllegalArgumentException("No putaway found for order ID ${params.id}")
         }
+        Location currentLocation = Location.get(session.warehouse.id)
+        LocationTypeCode[] locationTypeCodes = LocationTypeCode.BIN_LOCATION
+        List<Location> locations = locationService.getInternalLocations(currentLocation, locationTypeCodes, null)
```

Comment:
I think you should use putaway.origin instead of currentLocation (putaway should not be edited from different location, but we can use the origin just in case)